x-pack/metricbeat/module/openai/usage/usage_integration_test.go

```diff
@@ -0,0 +1,177 @@
+	events, errs := mbtest.ReportingFetchV2Error(f)
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, has %d: %v", len(errs), errs)
+	}
+
+	assert.NotEmpty(t, events)
+
+}
+
+func TestData(t *testing.T) {
+	apiKey := time.Now().String() // to generate a unique API key everytime
+	usagePath := "/usage"
+	server := initServer(usagePath, apiKey)
+	defer server.Close()
+
+	f := mbtest.NewReportingMetricSetV2Error(t, getConfig(server.URL+"/usage", apiKey))
+
+	err := mbtest.WriteEventsReporterV2Error(f, t, "")
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+}
+
+func getConfig(url, apiKey string) map[string]interface{} {
+	return map[string]interface{}{
+		"module":     "openai",
+		"metricsets": []string{"usage"},
+		"enabled":    true,
+		"period":     "1h",
+		"api_url":    url,
+		"api_keys": []map[string]interface{}{
+			{"key": apiKey},
+		},
+		"rate_limit": map[string]interface{}{
+			"limit": 60,
+			"burst": 5,
+		},
+		"collection": map[string]interface{}{
+			"lookback_days": 1,
+		},
+	}
+}
+
+func initServer(endpoint string, api_key string) *httptest.Server {
+	data := []byte(`{
+  "object": "list",
+  "data": [
+    {
+      "organization_id": "org-dummy",
+      "organization_name": "Personal",
+      "aggregation_timestamp": 1730696460,
+      "n_requests": 1,
+      "operation": "completion-realtime",
+      "snapshot_id": "gpt-4o-realtime-preview-2024-10-01",
+      "n_context_tokens_total": 118,
+      "n_generated_tokens_total": 35,
+      "email": null,
+      "api_key_id": null,
+      "api_key_name": null,
+      "api_key_redacted": null,
+      "api_key_type": null,
+      "project_id": null,
+      "project_name": null,
+      "request_type": "",
+      "n_cached_context_tokens_total": 0
+    },
+    {
+      "organization_id": "org-dummy",
+      "organization_name": "Personal",
+      "aggregation_timestamp": 1730696460,
+      "n_requests": 1,
+      "operation": "completion",
+      "snapshot_id": "gpt-4o-2024-08-06",
+      "n_context_tokens_total": 31,
+      "n_generated_tokens_total": 12,
+      "email": null,
+      "api_key_id": null,
+      "api_key_name": null,
+      "api_key_redacted": null,
+      "api_key_type": null,
+      "project_id": null,
+      "project_name": null,
+      "request_type": "",
+      "n_cached_context_tokens_total": 0
+    },
+    {
+      "organization_id": "org-dummy",
+      "organization_name": "Personal",
+      "aggregation_timestamp": 1730697540,
+      "n_requests": 1,
+      "operation": "completion",
+      "snapshot_id": "ft:gpt-3.5-turbo-0125:personal:yay-renew:APjjyG8E:ckpt-step-84",
+      "n_context_tokens_total": 13,
+      "n_generated_tokens_total": 9,
+      "email": null,
+      "api_key_id": null,
+      "api_key_name": null,
+      "api_key_redacted": null,
+      "api_key_type": null,
+      "project_id": null,
+      "project_name": null,
+      "request_type": "",
+      "n_cached_context_tokens_total": 0
+    }
+  ],
+  "ft_data": [],
+  "dalle_api_data": [],
```

Comment:
It'd be good to have data for each data set

Reply:
Yeah, I tried generating ft (fine-tuning) data but it doesn't seem to work. As OpenAI provides this API as undocumented, I couldn't find a single source with any samples. Not even sure they even populate for the response of this particular endpoint. For dalle_api_data, I'll add.